WcaOnRails/app/views/competitions/_admin_index_table.html.erb

```diff
@@ -0,0 +1,106 @@
+        Date
+      </th>
+      <th class="admin-date">
+        Announced
+      </th>
+      <th class="admin-date">
+        Report posted
+      </th>
+      <th class="admin-date">
+        Results posted
+      </th>
+      <th class="admin-button">
+        <span class="hide-div">(<%= tbody_display == "display: none;" ? "show" : "hide" %>)</span>
+      </th>
+    </tr>
+  </thead>
+  <tbody style="<%= tbody_display %>">
+    <% competitions.each_with_index do |competition, index| %>
+      <% announced = competition.announced_at %>
+      <% report = competition.delegate_report.posted_at %>
+      <% results = competition.results_posted_at %>
+      <tr>
+        <td>
+          <i class="flag f16 <%= competition.country.iso2.downcase %>"></i><%= link_to competition.cellName, competition_path(competition) %>
+          <br /><strong><%= competition.country_name %></strong>, <%= competition.cityName %>
+        </td>
+        <td class="admin-delegate">
+          <%= competition.delegates.map(&:name).to_sentence %>
+        </td>
+        <td class="admin-date">
+          <%= wca_date_range(competition.start_date, competition.end_date) %>
+        </td>
+        <% if announced %>
+          <% distance_announced = (competition.start_date - announced.to_date).to_i %>
+          <% if distance_announced <= 21 %>
+            <% announced_class = "alert-danger" %>
+          <% elsif distance_announced <= 28 %>
+            <% announced_class = "alert-orange" %>
+          <% else %>
+            <% announced_class = "alert-green" %>
+          <% end %>
+        <% end %>
+        <td class="admin-date <%= announced_class %>">
+          <% if announced %>
+            <%= announced.strftime("%b %d, %Y") %> (<%= distance_announced %>)
+          <% end %>
+        </td>
+
+        <% if report %>
+          <% distance_report = (report.to_date - competition.end_date).to_i %>
+          <% if distance_report > 21 %>
+            <% report_class = "alert-danger" %>
+          <% elsif distance_report > 14 %>
+            <% report_class = "alert-orange" %>
+          <% elsif distance_report > 7 %>
+            <% report_class = "alert-success" %>
+          <% else %>
+            <% report_class = "alert-green" %>
+          <% end %>
+        <% end %>
+        <td class="admin-date <%= report_class %>">
+          <% if report %>
+            <%= report.strftime("%b %d, %Y") %> (<%= distance_report %>)
+          <% end %>
+        </td>
+
+        <% if results %>
+          <% distance_results = (results.to_date - competition.end_date).to_i %>
+          <% if distance_results > 21 %>
+            <% results_class = "alert-danger" %>
+          <% elsif distance_results > 14 %>
+            <% results_class = "alert-orange" %>
+          <% elsif distance_results > 7 %>
+            <% results_class = "alert-success" %>
+          <% else %>
+            <% results_class = "alert-green" %>
+          <% end %>
+        <% end %>
```

Comment:
Please move these three if chains above to `competitions_helper.rb` to move this logic out of view.
Also we can simplify this
```ruby
results_class = if ...
                  "..."
                elsif ...
                  "..."
                else
                  "..."
                end
```